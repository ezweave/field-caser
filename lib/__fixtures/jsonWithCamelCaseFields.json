{
	"firstName": "Joe",
	"lastName": "Consumer",
	"details": {
		"dateOfBirth": "1/1/1970",
		"cars": [{
			"brand": "Pontiac",
			"model": "Aztek",
			"yearOfManufacture": "2004"
		},{
			"brand": "Saturn",
			"model": "Vue",
			"yearOfManufacture": "2004"
		}]
	}
}
