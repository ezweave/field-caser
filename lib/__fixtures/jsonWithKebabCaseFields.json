{
	"first-name": "Joe",
	"last-name": "Consumer",
	"details": {
		"date-of-birth": "1/1/1970",
		"cars": [{
				"brand": "Pontiac",
				"model": "Aztek",
				"year-of-manufacture": "2004"
			},{
				"brand": "Saturn",
				"model": "Vue",
				"year-of-manufacture": "2004"
		}]
	}
}
