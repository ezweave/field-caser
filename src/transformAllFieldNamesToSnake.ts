import { snakeCase } from 'lodash'
import { deepFieldTransformer } from './deepFieldTransformer'

export const transformAllFieldNamesToSnake: <T>(source: unknown) => T = deepFieldTransformer(snakeCase)