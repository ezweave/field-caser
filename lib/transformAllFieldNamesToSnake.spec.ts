import { transformAllFieldNamesToSnake } from './transformAllFieldNamesToSnake'
import * as jsonWithCamelCaseFields from './__fixtures/jsonWithCamelCaseFields.json'
import * as jsonWithKebabCaseFields from './__fixtures/jsonWithKebabCaseFields.json'

describe('transformAllFieldNamesToSnake', () => {
  it('transforms camel field names to snakecase', () => {
    expect(transformAllFieldNamesToSnake(jsonWithCamelCaseFields)).toMatchSnapshot()
  })
  it('transforms kebab field names to snakecase', () => {
    expect(transformAllFieldNamesToSnake(jsonWithKebabCaseFields)).toMatchSnapshot()
  })
})