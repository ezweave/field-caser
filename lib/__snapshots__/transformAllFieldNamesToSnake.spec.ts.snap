// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformAllFieldNamesToSnake transforms camel field names to snakecase 1`] = `
Object {
  "default": Object {
    "details": Object {
      "cars": Array [
        Object {
          "brand": "Pontiac",
          "model": "Aztek",
          "year_of_manufacture": "2004",
        },
        Object {
          "brand": "Saturn",
          "model": "Vue",
          "year_of_manufacture": "2004",
        },
      ],
      "date_of_birth": "1/1/1970",
    },
    "first_name": "Joe",
    "last_name": "Consumer",
  },
  "details": Object {
    "cars": Array [
      Object {
        "brand": "Pontiac",
        "model": "Aztek",
        "year_of_manufacture": "2004",
      },
      Object {
        "brand": "Saturn",
        "model": "Vue",
        "year_of_manufacture": "2004",
      },
    ],
    "date_of_birth": "1/1/1970",
  },
  "first_name": "Joe",
  "last_name": "Consumer",
}
`;

exports[`transformAllFieldNamesToSnake transforms kebab field names to snakecase 1`] = `
Object {
  "default": Object {
    "details": Object {
      "cars": Array [
        Object {
          "brand": "Pontiac",
          "model": "Aztek",
          "year_of_manufacture": "2004",
        },
        Object {
          "brand": "Saturn",
          "model": "Vue",
          "year_of_manufacture": "2004",
        },
      ],
      "date_of_birth": "1/1/1970",
    },
    "first_name": "Joe",
    "last_name": "Consumer",
  },
  "details": Object {
    "cars": Array [
      Object {
        "brand": "Pontiac",
        "model": "Aztek",
        "year_of_manufacture": "2004",
      },
      Object {
        "brand": "Saturn",
        "model": "Vue",
        "year_of_manufacture": "2004",
      },
    ],
    "date_of_birth": "1/1/1970",
  },
  "first_name": "Joe",
  "last_name": "Consumer",
}
`;
